week-4/E-arrays-of-objects/exercise-2.js

```diff
@@ -7,50 +7,92 @@ Each destination has a name, a distance from Glasgow, and a list of transportati
 3) Print in the console all the destination names more than 300 kms far away and reachable by train.
 */
 
-
 var destination1 = {
-    destinationName: "Edinburgh",
-    distanceKms: 80,
-    transportations: ["car", "bus", "train"]
+  destinationName: "Edinburgh",
+  distanceKms: 80,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination2 = {
-    destinationName: "London",
-    distanceKms: 650,
-    transportations: ["car", "bus", "train"]
+  destinationName: "London",
+  distanceKms: 650,
+  transportations: ["car", "bus", "train"]
 };
 
 var destination3 = {
-    destinationName: "Paris",
-    distanceKms: 900,
-    transportations: ["train", "plane"]
+  destinationName: "Paris",
+  distanceKms: 900,
+  transportations: ["train", "plane"]
 };
 
 var destination4 = {
-    destinationName: "Dublin",
-    distanceKms: 350,
-    transportations: ["plane", "ferry"]
+  destinationName: "Dublin",
+  distanceKms: 350,
+  transportations: ["plane", "ferry"]
 };
 
-var travelDestinations = [destination1, destination2, destination3, destination4];
+var travelDestinations = [
+  destination1,
+  destination2,
+  destination3,
+  destination4
+];
 
 /* 
 DO NOT EDIT ANYTHING ABOVE THIS LINE
 WRITE YOUR CODE BELOW
 */
+//Question 1
+// Find retorna só o primeiro que satisfaz a condiçao. Filter retorna mais!
+
+function checkingDistance(travelDestinations) {
+  if (travelDestinations.distanceKms <= 500) {
+    return travelDestinations;
+  }
+}
```

Comment:
Some problems here.  
checkingDistance() only ever takes ONE object at a time.  YES, it will be called repeatedly by filter, but each time it is called, it takes ONE travel destination, so it's parameter should not be plural - it does not take an array of travelDestinations.

Secondly, what you want to return for a function used by filter (a "predicate function") is simply true or false.  You can (and should) convince yourself of this by reading the official docs: 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter#Syntax
"Function is a predicate, to test each element of the array. Return true to keep the element, false otherwise."

Thirdly, you again have a path through your code where you probably can't tell me what will be returned.

Fourthly, there's an easier way to do this without using an `if` statement at all.